src/clj/fluree/db/api/query.cljc

```diff
@@ -43,13 +47,20 @@
 (defn restrict-db
   [db t context opts]
   (go-try
-    (let [db*  (if-let [policy-identity (perm/parse-policy-identity opts context)]
-                 (<? (perm/wrap-policy db policy-identity))
-                 db)
-          db** (-> (if t
-                     (<? (time-travel/as-of db* t))
-                     db*))]
-      (assoc-in db** [:policy :cache] (atom {})))))
+    (let [policy-db      (if-let [policy-identity (perm/parse-policy-identity opts context)]
+                           (<? (perm/wrap-policy db policy-identity))
+                           db)
+          time-travel-db (-> (if t
+                               (<? (time-travel/as-of policy-db t))
+                               policy-db))
+          reasoned-db    (let [{:keys [reasoners reasoner-rules reasoner-rules-db]} opts]
+                           (if reasoners
+                             ;; Currently we only support one rule source, so we take the first db or first
+                             ;; reason graph that we find.
+                             (<? (reasoner/reason
+                                  time-travel-db reasoners (or (first reasoner-rules-db) (first reasoner-rules)) opts))
```

Comment:
Can you split this form up into multiple lines so it doesn't extend so far to the right?